x-pack/plugins/security/server/lib/authorization/check_privileges.js

```diff
@@ -59,13 +61,15 @@ export function checkPrivilegesWithRequestFactory(shieldClient, config, actions,
           }],
           index: [{
             names: [kibanaIndex],
-            privileges: ['create', 'delete', 'read', 'view_index_metadata']
+            privileges: buildLegacyIndexPrivileges()
           }],
         }
       });
 
+      validateEsPrivilegeResponse(hasPrivilegesResponse, application, allApplicationPrivileges, [ALL_RESOURCE], kibanaIndex);
```

Comment:
What if instead of introducing the `buildLegacyIndexPrivileges` abstraction, we passed the "index privileges" that we expect to be on the response to the `validateEsPrivilegeResource` function, similar to the way that we're passing the more dynamic `allApplicationPrivileges`?

Reply:
I originally had it that way. I came up with two reasons to try this other approach:
1. I didn't like the number of parameters this function was expecting, and one less felt a bit more sane to me
2. The abstraction allows me to build the legacy validation schema once, rather than building it on every request. This is probably unnecessary given the timings that I ran, but I was worried about performance when I put this together.

I can move it back if you'd still prefer -- I don't have a strong opinion here

Reply:
That makes sense to me :) I actually intended to delete this comment...